pkg/resources/kafka/kafka.go

```diff
@@ -942,6 +954,22 @@ func (r *Reconciler) getK8sAssignedNodeport(log logr.Logger, eListenerName strin
 	return nodePort, nil
 }
 
+// determineControllerId returns the ID of the controller broker of the current cluster
+func (r *Reconciler) determineControllerId() (int32, error) {
+	kClient, close, err := r.kafkaClientProvider.NewFromCluster(r.Client, r.KafkaCluster)
+	if err != nil {
+		return -1, errors.WrapIf(err, "could not create Kafka client, thus could not determine controller")
+	}
+	_, controllerID, err := kClient.DescribeCluster()
+	if err != nil {
+		return -1, errors.WrapIf(err, "could not find controller broker")
+	}
+
+	defer close()
```

Comment:
I'd move this up before the `kClient.DescribeCluster()`. In case the controller could not be determined, the client connections could leak.

Reply:
That is a good catch:) Or maybe this method just returns an int32 and logs error instead of returning.

Reply:
Nice catch.